src/universal/redux/segmentActions.js

```diff
@@ -1,22 +1,75 @@
 import {EventTypes} from 'redux-segment';
+import {cashay} from 'cashay';
+import {getAuthQueryString, getAuthedOptions} from './getAuthedUser';
 
 /**
  * Sometimes, we just want to track events in segment.
  */
 
 const SEGMENT_EVENT = '@@segment/EVENT';
 
-export function segmentEvent(event, properties) {
-  return {
-    type: SEGMENT_EVENT,
-    meta: {
-      analytics: {
-        eventType: EventTypes.track,
-        eventPayload: {
-          event,
-          properties
+const defaultProfile = {
+  avatar: null,
+  email: null,
+  id: null,
+  name: null
+};
+
+export function selectSegmentProfile(state) {
+  if (!state.auth || !cashay.store) { return defaultProfile; }
+  const userId = state.auth.obj.sub;
+  const {user} = cashay.query(getAuthQueryString, getAuthedOptions(userId)).data;
+  if (!user) { return defaultProfile; }
```

Comment:
if `state.auth.obj.sub` exists, then the `cashay.entities.User` will exist in the same tick (albeit with nulls until the async req comes back). That means we could say `if (!userId) { return defaultProfile; }` before the query.